### PR TITLE
txn: fix the maximum value of a transaction from 10GB to 1 TiB

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [x] v8.3 (TiDB 8.3 versions)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-transaction-restraints.md [hunk 1]
```diff
@@ -712,8 +712,15 @@ mysql> SELECT * FROM T2;
...
-- 支持的最大单个事务容量为 10GB（TiDB v4.0 及更高版本可通过 tidb-server 配置项 `performance.txn-total-size-limit` 调整，低于 TiDB v4.0 的版本支持的最大单个事务容量为 100MB）。
+- 最大单行记录容量为 120 MiB。
+
+    - TiDB v4.0.10 及更高的 v4.0.x 版本、v5.0.0 及更高的版本可通过 tidb-server 配置项 [`performance.txn-entry-size-limit`](/tidb-configuration-file.md#txn-entry-size-limit-从-v4010-和-v500-版本开始引入) 调整，低于 TiDB v4.0.10 的版本支持的单行容量为 6 MiB。
+    - 从 v7.6.0 开始，你可以使用 [`tidb_txn_entry_size_limit`](/system-variables.md#tidb_txn_entry_size_limit-从-v760-版本开始引入) 系统变量动态修改该配置项的值。
+
+- 支持的最大单个事务容量为 1 TiB。
+
+    - TiDB v4.0 及更高版本可通过 tidb-server 配置项 `performance.txn-total-size-limit` 调整，低于 TiDB v4.0 的版本支持的最大单个事务容量为 100 MiB。
+    - 在 v6.5.0 及之后的版本中，不再推荐使用该配置项，事务的内存大小会被累计计入所在会话的内存使用量中，并由 [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query) 变量在单个会话内存超阈值时采取控制行为。
```
Comment (hfxsd) [inline develop/dev-guide-transaction-restraints.md:723]:
```suggestion
    - 在 v6.5.0 及之后的版本中，不再推荐使用配置项 `performance.txn-total-size-limit`，事务的内存大小会被累计计入所在会话的内存使用量中，并由 [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query) 变量在单个会话内存超阈值时采取控制行为。
```

File: releases/release-4.0.0-rc.2.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ TiDB 版本：4.0.0-rc.2
 
 + TiDB
 
-    - 去掉了特别为开启 Binlog 时定义的事务容量上限 (100 MB)，现在事务的容量上限统一为 10 GB，但若开启 Binlog 且下游是 Kafka，由于 Kafka 消息大小的限制是 1 GB，请根据情况调整 `txn-total-size-limit` 配置参数 [#16941](https://github.com/pingcap/tidb/pull/16941)
```
Comment (ekexium) [inline releases/release-4.0.0-rc.2.md:17]: 这个 4.0 的 release note 需要改成 1 TiB 吗？应该也是 10 GiB 吧

Comment (hfxsd) [reply releases/release-4.0.0-rc.2.md:17]: @crazycs520 PTAL

File: develop/dev-guide-transaction-restraints.md [hunk 3]
```diff
@@ -712,8 +712,15 @@ mysql> SELECT * FROM T2;
...
-- 支持的最大单个事务容量为 10GB（TiDB v4.0 及更高版本可通过 tidb-server 配置项 `performance.txn-total-size-limit` 调整，低于 TiDB v4.0 的版本支持的最大单个事务容量为 100MB）。
+- 最大单行记录容量为 120 MiB。
+
+    - TiDB v4.0.10 及更高的 v4.0.x 版本、v5.0.0 及更高的版本可通过 tidb-server 配置项 [`performance.txn-entry-size-limit`](/tidb-configuration-file.md#txn-entry-size-limit-从-v4010-和-v500-版本开始引入) 调整，低于 TiDB v4.0.10 的版本支持的单行容量为 6 MiB。
+    - 从 v7.6.0 开始，你可以使用 [`tidb_txn_entry_size_limit`](/system-variables.md#tidb_txn_entry_size_limit-从-v760-版本开始引入) 系统变量动态修改该配置项的值。
+
+- 支持的最大单个事务容量为 1 TiB。
+
+    - TiDB v4.0 及更高版本可通过 tidb-server 配置项 `performance.txn-total-size-limit` 调整，低于 TiDB v4.0 的版本支持的最大单个事务容量为 100 MiB。
+    - 在 v6.5.0 及之后的版本中，不再推荐使用配置项 `performance.txn-total-size-limit`，事务的内存大小会被累计计入所在会话的内存使用量中，并由 [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query) 变量在单个会话内存超阈值时采取控制行为。
```
Comment (crazycs520) [inline develop/dev-guide-transaction-restraints.md:723]: 这里感觉有点歧义，要不加一个 performance.txn-total-size-limit 参数文档的 link 跳过去吧。同样的内容不用在这里重复一遍？
大于 v6.5.0 版本中，tidb_mem_quota_query 生效的前提是没有设置performance.txn-total-size-limit 的值。

File: releases/release-4.0.0-rc.2.md [hunk 4]
```diff
@@ -14,7 +14,7 @@ TiDB 版本：4.0.0-rc.2
...
+    - 去掉了特别为开启 Binlog 时定义的事务容量上限 (100 MB)，现在事务的容量上限统一为 1 TiB，但若开启 Binlog 且下游是 Kafka，由于 Kafka 消息大小的限制是 1 GiB，请根据情况调整 `txn-total-size-limit` 配置参数 [#16941](https://github.com/pingcap/tidb/pull/16941)
```
Comment (hfxsd) [inline releases/release-4.0.0-rc.2.md:17]:
```suggestion
    - 去掉了特别为开启 Binlog 时定义的事务容量上限 (100 MB)，现在事务的容量上限统一为 10 GB，但若开启 Binlog 且下游是 Kafka，由于 Kafka 消息大小的限制是 1 GB，请根据情况调整 `txn-total-size-limit` 配置参数 [#16941](https://github.com/pingcap/tidb/pull/16941)
```

**Review by crazycs520 — APPROVED** [review]

File: basic-features.md [hunk 5]
```diff
@@ -88,6 +88,8 @@ aliases: ['/docs-cn/dev/basic-features/','/docs-cn/dev/experimental-features-4.0
 | [`ALTER TABLE ... COMPACT`](/sql-statements/sql-statement-alter-table-compact.md) | Y |  Y |  Y |  Y  | Y | Y | E | N | N | N | N |
 | [表级锁 (Table Lock)](/sql-statements/sql-statement-lock-tables-and-unlock-tables.md) | E |  E  |  E  |  E  | E | E | E | E | E | E | E |
 | [物化列式存储的查询结果](/tiflash/tiflash-results-materialization.md) | Y |  Y |  Y |  Y  | Y | E | N | N | N | N | N |
+| [标准 DML 语句的执行方式 (`tidb_dml_type = "standard"`)](/system-variables.md#tidb_dml_type-从-v800-版本开始引入) | Y |  Y | Y |  N | N | N | N | N | N | N | N |
```
Comment (hfxsd) [inline basic-features.md:91]:
```suggestion
```

**Review by ekexium — APPROVED** [review]

File: develop/dev-guide-transaction-restraints.md (hunk 3 above)
Comment (hfxsd) [reply develop/dev-guide-transaction-restraints.md]: OK updated

**Pull request merged**